riscof-plugin/sail_cSim/env/link.ld

```diff
@@ -0,0 +1,18 @@
+OUTPUT_ARCH( "riscv" )
```

Comment:
There are going to be lots of different platforms out there that want this same linker script, or very similar versions to it (QEMU is going to be identical, as is Spike). I don't think this should be duplicated to every platform, there should probably be just one file inside riscof itself.

Reply:
There will be even more that have different linker scripts, and we have examples of those.
The linker script is customized to every model - we can make no assumptions what the platform looks like. 
The fact that some models will be the same (you happened to choose software models as opposed to RTL, which makes it more likely) doesn't change the reality that there will be a multitude of linker scripts, and they won't be hosted in one place, but hosted by the model itself,

Reply:
So why can there not just be a single "software models" linker script in riscof? This is just promoting unnecessary copy/paste development practices.

Reply:
Riscof has not idea whether a model is a SW model or not, and doesn't want to. It takes two models and compares them.

Reply:
It doesn't need to; it can just provide a set of common implementations that platforms can optionally use. I just don't want to see three copies of the same files in sail-riscv, QEMU and Spike, which is what is currently going to happen and is a sign that the current approach could be improved.

Reply:
Also, have you considered specifying a standard interface for these things rather than requiring platform-specific code in the first place? For example, making use of the semihosting spec? That seems like it would be ideal for some of this.

Reply:
Linker scripts support a `INCLUDE` directive that could be used to avoid lots of copy-pasting. Having a few common .ld files that can be include will avoid lots of divergence and will make it easier to add new features.

Our previous CHERI-MIPS testsuite had various linker very similar scripts with slightly different parameters (e.g. running in cached vs uncached memory). Having all those scripts made the switch from ld.bfd to ld.lld extremely tedious since I had to look at all these very similar scripts and figure out from the git history which differences were actually indented and which of them were just cases of "this commit forgot to update 2 of the scripts but should have done so".

If riscof provides something like a `default-params.ld`
```
__base_addr__ = 0x80000000;
__default_align__ = 0x1000;

OUTPUT_ARCH( "riscv" )
ENTRY(rvtest_entry_point)
```
and a `default-sections.ld`
```
SECTIONS
{
  . = __base_addr__;
  .text.init : { *(.text.init) }
  . = ALIGN(__default_align__);
  .tohost : { *(.tohost) }
  . = ALIGN(__default_align__);
  .text : { *(.text) }
  . = ALIGN(__default_align__);
  .data : { *(.data) }
  .data.string : { *(.data.string)}
  .bss : { *(.bss) }
  _end = .;
}
```
Then all that sail and QEMU need to do would be something like
```
INCLUDE default-params.ld
INCLUDE default-sections.ld
```
(or just provide that as some software-model default linker script)

And to use a different start address you can just use 
```
INCLUDE default-params.ld
/* Override base address */
__base_addr__ = 0xc0000000;
INCLUDE default-sections.ld
```